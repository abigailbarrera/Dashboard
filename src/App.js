import React from 'react';
import { BrowserRouter, Route, Switch } from 'react-router-dom';
import { Dashboard } from './components/Dashboard';
import { Addcampaign } from './components/AddCampaign';
import { Editcampaign } from './components/EditCampaign';

import { GlobalProvider } from './context/GlobalState';

export const App = () => {
  return (
    <GlobalProvider>
      <BrowserRouter>
        <Switch>
          <Route path="/" component={Dashboard} exact />
          <Route path="/add" component={Addcampaign} exact />
          <Route path="/edit/:id" component={Editcampaign} exact />
        </Switch>
      </BrowserRouter>
    </GlobalProvider>
  );
}
