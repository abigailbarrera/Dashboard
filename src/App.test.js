import React from 'react';
import { create } from 'react-test-renderer';
 
import { App } from './App';
 
describe('App', () => {
  it('snapshot renders', () => {
    let tree = create(<App />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
