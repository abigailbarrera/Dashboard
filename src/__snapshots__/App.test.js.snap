// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot renders 1`] = `
<div
  className="sc-AxjAm dAHyqm"
>
  <h1
    className="sc-AxirZ jfryYJ"
  >
    Dashboard
  </h1>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-12 mt-5"
      >
        <table
          className="table table-striped table-bordered table-hover"
        >
          <thead>
            <tr>
              <th>
                Live (sent) Campaigns
              </th>
              <th>
                Amount Sent
              </th>
              <th>
                Click Through Rates
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>
                Fourth of July Promo
              </td>
              <td>
                6506
              </td>
              <td>
                6137
              </td>
            </tr>
            <tr>
              <td>
                Labor Day Sale
              </td>
              <td>
                7228
              </td>
              <td>
                6883
              </td>
            </tr>
            <tr>
              <td>
                Cotton Candy Palette Announcement
              </td>
              <td>
                7246
              </td>
              <td>
                7080
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-12 mt-5"
      >
        <table
          className="table table-striped table-bordered table-hover"
        >
          <thead>
            <tr>
              <th>
                <a
                  href="/add"
                  onClick={[Function]}
                >
                  <button
                    className="btn btn-light btn-sm m-0"
                    type="button"
                  >
                    Create a new Campaign
                  </button>
                </a>
              </th>
              <th>
                Open (unsent) Campaigns
              </th>
              <th>
                Target Segment
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>
                <a
                  href="/edit/4"
                  onClick={[Function]}
                >
                  <button
                    className="btn btn-light btn-sm m-0"
                    onClick={[Function]}
                    type="button"
                  >
                    Edit Details
                  </button>
                </a>
              </td>
              <td>
                Fall Decor Updates
              </td>
              <td>
                Decor Purchases
              </td>
            </tr>
            <tr>
              <td>
                <a
                  href="/edit/5"
                  onClick={[Function]}
                >
                  <button
                    className="btn btn-light btn-sm m-0"
                    onClick={[Function]}
                    type="button"
                  >
                    Edit Details
                  </button>
                </a>
              </td>
              <td>
                Great Halloween Sale
              </td>
              <td>
                All Subscribers
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;
